recipes/entt/3.x.x/conanfile.py

```diff
@@ -0,0 +1,86 @@
+import os
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class ConanRecipe(ConanFile):
+    name = "entt"
+
+    description = "Gaming meets modern C++ - a fast and reliable entity-component system (ECS) and much more"
+    topics = ("conan," "entt", "gaming", "entity", "ecs")
+
+    homepage = "https://github.com/skypjack/entt"
+    url = "https://github.com/conan-io/conan-center-index"
+
+    license = "MIT"
+
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def configure(self):
+        version = tools.Version(self.settings.compiler.version)
+        compiler = str(self.settings.compiler)
+
+        if self.settings.compiler.cppstd:
+            tools.check_min_cppstd(self, "17")
+            return
```

Comment:
What's the reason to return here? Shouldn't we also validate for a minimum required compiler version? What if someone decides to use gcc6 with cppstd=17?
(btw, nice idea declaring the minimal_version as a dictionary, I'll use it on other recipes :-P)

Reply:
Using an unsupported standard version on an old compiler is a broken invariant and is already reported by Conan.
Example:
> The specified 'compiler.cppstd=20' is not available for 'apple-clang 10.0'. Possible values are ['11', '14', '17', '98', 'gnu11', 'gnu14', 'gnu17', 'gnu98']'

Reply:
Sorry, maybe I didnt explain myself well. After reading your recipe I thougth, `entt` developer stated that compilers no less than gcc-7, clang-5, apple-clang-10, etc... should be used. Then I paid attention to the `return` after `check_min_cppstd` which implies that any compiler that is C++17 will do. 
Notice my previous example with gcc-6 (cppstd=17 is allowed on gcc 6)
> ERROR: The specified 'compiler.cppstd=20' is not available for 'gcc 6.3'. Possible values are ['11', '14', '17', '98', 'gnu11', 'gnu14', 'gnu17', 'gnu98']'

So, my question is: Is that minimal_version just a fallback in case `cppstd = None` or the library author specifically requires those versions?

Reply:
I see now, thanks for pointing it out. 

So, the idea is that any compiler that supports C++17 will do for us, yes. If you set the gcc6 and the cppstd=17 it seems to allow it. The `None` value checks the default standard version which is at most C++14 on newer versions.

I start to think that the `check_min_cppstd` is not the best way because gcc6 reports that it supports the C++17, but in practice, it lacks a ton of features until the gcc7 at least. I'm not sure what we can do about that.

Reply:
I'll rearrange the checks. I'm not sure are they're required by the author, but they were there before me. 🙂

Reply:
Oh wait, don't mind the previous comment. I took them from another package supporting at least C++17.